classes/salesforce_pull.php

```diff
@@ -650,15 +650,6 @@ function check_user( $pull_allowed, $object_type, $object, $sf_sync_trigger, $sa
 
 			if ( $is_new === true && ( $sf_sync_trigger == $this->mappings->sync_sf_create ) ) {
 
-				// right here we should set the pulling transient
```

Comment:
this chunk was creating an object map whether it was a create or upsert, but was only adding the correct wordpress id if it was a create - if a new object was created.